### PR TITLE
CB-1321: Detect SQL Injection

### PR DESCRIPTION
**Pull request by joshuadiablito**

### CODE REVIEW

File: test/routes/middlewares/contains-sql.test.js [hunk 1]
```diff
@@ -0,0 +1,79 @@
+const sinon = require('sinon');
+
+const containsSQLMiddleware = require('../../../app/routes/middlewares/contains-sql');
+
+describe('containsSQLMiddleware', () => {
```
Comment (p3dr0migue1) [inline test/routes/middlewares/contains-sql.test.js]: We should make clear what is being tested, a suggestion:
```js
describe('Test authMiddleware')
```
This also ensures its consistent across the project

File: test/routes/middlewares/contains-sql.test.js [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+  describe('when it is called', () => {
+    describe('and there is an originalUrl', () => {
+      describe('and it contains SQL', () => {
+        it('should call next and store the decoded token', () => {
```
Comment (p3dr0migue1) [inline test/routes/middlewares/contains-sql.test.js]: I find this structure slighly hard to follow, why not keep it simple and just do:

```js
describe('Test authMiddleware')
  it('should remove all sql injection code from the url')
    // test
```

File: test/routes/middlewares/contains-sql.test.js [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+          const request = { originalUrl };
+          const status = sinon.stub().returns({ json });
+          const response = { status, locals: { user: null } };
+          const next = sinon.spy();
+
+          containsSQLMiddleware(request, response, next);
+
+          sinon.assert.calledWith(status, 403);
+          sinon.assert.calledWith(json, { error: 'Unauthorized' });
+          sinon.assert.notCalled(next);
```
Comment (p3dr0migue1) [inline test/routes/middlewares/contains-sql.test.js]: the description says `it('should call next..` but here the assert, asserts that is not called

File: test/routes/middlewares/contains-sql.test.js [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+
+const containsSQLMiddleware = require('../../../app/routes/middlewares/contains-sql');
+
+describe('containsSQLMiddleware', () => {
+  describe('when it is called', () => {
+    describe('and there is an originalUrl', () => {
+      describe('and it contains SQL', () => {
+        it('should call next and store the decoded token', () => {
+          const json = sinon.spy();
+          const originalUrl = 'http://localhost/my/cool/url/has?=\'--SELECT%20some,%20fields';
```
Comment (p3dr0migue1) [inline test/routes/middlewares/contains-sql.test.js]: since you are not matching new and old urls you can just rename this const to `url` instead of `originalUrl` 
probably worth updating the example to a slightly more realistic example:
```
http://localhost:8080/v2/roles?filter=name=eq.Manager\'--SELECT%20some,%20fields';
```
this will help us down the line to understand the security implemented when reading the code

File: test/routes/middlewares/contains-sql.test.js (hunk 2 above)
Comment (joshuadiablito) [reply test/routes/middlewares/contains-sql.test.js]: oops - I needed to update the text.. copied from other test

File: test/routes/middlewares/contains-sql.test.js (hunk 1 above)
Comment (joshuadiablito) [reply test/routes/middlewares/contains-sql.test.js]: Sounds good - but not authMiddleware! :)

File: test/routes/middlewares/contains-sql.test.js (hunk 3 above)
Comment (joshuadiablito) [reply test/routes/middlewares/contains-sql.test.js]: yep forgot to update them, good catch thanks

File: test/routes/middlewares/contains-sql.test.js (hunk 4 above)
Comment (joshuadiablito) [reply test/routes/middlewares/contains-sql.test.js]: sounds good :)

Comment (joshuadiablito) [reply test/routes/middlewares/contains-sql.test.js]: I made it `originalUrl` as the check is against originalUrl, so we make use of destructuring/restructuring rather than having to write `{ originalUrl: url }` instead of `{ originalUrl }`

File: test/routes/middlewares/contains-sql.test.js (hunk 2 above)
Comment (joshuadiablito) [reply test/routes/middlewares/contains-sql.test.js]: and the BDD syntax allows us to see what is tested and what isn't tested, allows us to see "he has checked that there is SQL, and that there is NO sql".

Comment (joshuadiablito) [reply test/routes/middlewares/contains-sql.test.js]: Also, allows non-developers to read the tests and understand what is tested/not tested without getting lost in the implementation

Comment (joshuadiablito) [reply test/routes/middlewares/contains-sql.test.js]: also, when running the tests you get to see a nice BDD output:
```
  Test containsSQLMiddleware
    when it is called
      and there is an originalUrl
        and it contains SQL
          ✓ should NOT call next and should return Unauthorized message
        and it does NOT contain SQL
          ✓ should call next and NOT return Unauthorized message
      and there is a falsy originalUrl
        and the value is null
          ✓ should call next and NOT return Unauthorized message
        and the value is undefined
          ✓ should call next and NOT return Unauthorized message
```
